packages/flutter/test/cupertino/magnifier_test.dart

```diff
@@ -0,0 +1,259 @@
+        return Offset(
+            animatedPositioned.left ?? 0, animatedPositioned.top ?? 0);
+      }
+
+      testWidgets('should be at gesture position if does not violate any positioning rules', (WidgetTester tester) async {
+        final Key fakeTextFieldKey = UniqueKey();
+        final Key outerKey = UniqueKey();
+
+        await tester.pumpWidget(
+          Container(
+            key: outerKey,
+            color: const Color.fromARGB(255, 0, 255, 179),
+            child: MaterialApp(
+              home: Center(
+                child: Container(
+                  key: fakeTextFieldKey,
+                  width: 10,
+                  height: 10,
+                  color: Colors.red,
+                  child: const Placeholder(),
+                ),
+              ),
+            ),
+          ),
+        );
+        final BuildContext context = tester.element(find.byType(Placeholder));
+
+        // Magnifier should be positioned directly over the red square.
+        final RenderBox tapPointRenderBox =
+            tester.firstRenderObject(find.byKey(fakeTextFieldKey)) as RenderBox;
+        final Rect fakeTextFieldRect =
+            tapPointRenderBox.localToGlobal(Offset.zero) & tapPointRenderBox.size;
+
+        final ValueNotifier<MagnifierOverlayInfoBearer> magnifier =
+            ValueNotifier<MagnifierOverlayInfoBearer>(
+          MagnifierOverlayInfoBearer(
+            currentLineBoundries: fakeTextFieldRect,
+            fieldBounds: fakeTextFieldRect,
+            caretRect: fakeTextFieldRect,
+            // The tap position is dragBelow units below the text field.
+            globalGesturePosition: fakeTextFieldRect.center,
+          ),
+        );
+
+        await showCupertinoMagnifier(context, tester, magnifier);
+
+        // Should show two red squares; original, and one in the magnifier,
+        // directly ontop of one another.
+        await expectLater(
+          find.byKey(outerKey),
+          matchesGoldenFile('cupertino_magnifier.position.default.png'),
+        );
+      });
+
+      testWidgets('should never horizontally be outside of Screen Padding',
+          (WidgetTester tester) async {
+        await tester.pumpWidget(
+          const MaterialApp(
+            color: Color.fromARGB(7, 0, 129, 90),
+            home: Placeholder(),
+          ),
+        );
+
+        final BuildContext context = tester.firstElement(find.byType(Placeholder));
+
+        await showCupertinoMagnifier(
+            context,
+            tester,
+            ValueNotifier<MagnifierOverlayInfoBearer>(
+                MagnifierOverlayInfoBearer(
+              currentLineBoundries: reasonableTextField,
+              fieldBounds: reasonableTextField,
+              caretRect: reasonableTextField,
+              // The tap position is far out of the right side of the app.
+              globalGesturePosition:
+                  Offset(MediaQuery.of(context).size.width + 100, 0),
+            )));
```

Comment:
Nit: the indentation here is wonky (this applies to several tests below as well). It should be something like:

```suggestion
            ValueNotifier<MagnifierOverlayInfoBearer>(
               MagnifierOverlayInfoBearer(
                 currentLineBoundries: reasonableTextField,
                 fieldBounds: reasonableTextField,
                 caretRect: reasonableTextField,
                 // The tap position is far out of the right side of the app.
                 globalGesturePosition:  Offset(MediaQuery.of(context).size.width + 100, 0),
               )
             )
           );
```